src/app/Caches/Create/Features/IndexedCache.tsx

```diff
@@ -0,0 +1,166 @@
+        setIndexedEntities(newChips);
+    };
+
+    const addChip = (newChipText: string) => {
+        setIndexedEntities([...indexedEntities, `${newChipText}`]);
+        setEntityInput('');
+    };
+
+    const helperAddEntity = () => {
+        if (entityInput.length) {
+            if (!indexedEntities.includes(entityInput)) {
+                addChip(entityInput);
+                setValidEntity('success')
+            }
+            else {
+                setValidEntity('error');
+            }
+        }
+    }
+
+    /** enable keyboard only usage while focused on the text input */
+    const handleTextInputKeyDown = (event?: React.KeyboardEvent) => {
+        if (event && event.key === KEY_CODES.ESCAPE_KEY.toString()) {
+            helperAddEntity();
+        }
+    };
+
+    const handleAddButton = () => {
+        helperAddEntity();
+    }
+
+    const formIndexEntities = () => {
+        return (
+            <FormGroup isInline fieldId='indexed-entities' validated={validEntity} helperTextInvalid={t('caches.create.configurations.feature.index-storage-entity-helper')}>
+                <Stack hasGutter style={{
+                    width: '100%',
+                }}>
+                    <StackItem>
+                        <TextContent>
+                            <MoreInfoTooltip label={t('caches.create.configurations.feature.index-storage-entity')} toolTip={t('caches.create.configurations.feature.index-storage-entity-tooltip', { brandname: brandname })} textComponent={TextVariants.h3} />
+                        </TextContent>
+                    </StackItem>
+                  <StackItem>
+                    <InputGroup style={{ marginTop: global_spacer_sm.value }} >
+                      <TextInput value={entityInput} type="text" onChange={(val) => setEntityInput(val)} onKeyDown={handleTextInputKeyDown}/>
+                      <Button id="textAreaButton2" variant="control" onClick={handleAddButton}>
+                        Add
+                      </Button>
+                    </InputGroup>
+                  </StackItem>
+                    <StackItem>
+                        {indexedEntities.map(currentChip => (
+                            <Label
+                                color="blue"
+                                closeBtnAriaLabel="Custom close button for editable label"
+                                onEditComplete={(text) => {
+                                    setIndexedEntities(indexedEntities.map(chip => chip === currentChip ? text : chip));
+                                }}
+                                isEditable
+                                key={currentChip} onClose={() => deleteChip(currentChip)}>
+                                {currentChip}
+                            </Label>
+                        ))}
+                    </StackItem>
+                </Stack>
+            </FormGroup >
+        )
+    }
+
+    return (
+        <Form onSubmit={(e) => {
```

Comment:
@karesti Form submit issue is still there under index entities field, please review.